cosmos/runtime/runtime.go

```diff
@@ -21,123 +21,150 @@
 	p.WrappedTxPool = txpool.New(p.TxPool())
-	bApp.SetMempool(p.WrappedTxPool)
+	app.SetMempool(p.WrappedTxPool)
 
 	p.WrappedMiner = miner.New(p.Miner())
-	bApp.SetPrepareProposal(p.WrappedMiner.PrepareProposal)
+	app.SetPrepareProposal(p.WrappedMiner.PrepareProposal)
 
 	if err := ek.Setup(p.Blockchain()); err != nil {
 		return err
 	}
 
 	// Set the ante handler to nil, since it is not needed.
-	bApp.SetAnteHandler(nil)
+	app.SetAnteHandler(nil)
 
 	return nil
 }
 
-// Init is a function that initializes the Polaris struct.
-// It takes a client context and a logger as arguments.
-// It returns an error if the initialization fails.
-func (p *Polaris) Init(clientCtx client.Context, logger log.Logger, ek EVMKeeper) error {
-	// Initialize services.
+// SetupServices initializes and registers the services with Polaris.
+// It takes a client context as an argument and returns an error if the setup fails.
+func (p *Polaris) SetupServices(clientCtx client.Context) error {
+	// Initialize the miner with a new execution payload serializer.
 	p.WrappedMiner.Init(libtx.NewSerializer[*engine.ExecutionPayloadEnvelope](
 		clientCtx.TxConfig, evmtypes.WrapPayload))
 
-	p.WrappedTxPool.Init(logger, clientCtx, libtx.NewSerializer[*coretypes.Transaction](
+	// Initialize the txpool with a new transaction serializer.
+	p.WrappedTxPool.Init(p.logger, clientCtx, libtx.NewSerializer[*coretypes.Transaction](
 		clientCtx.TxConfig, evmtypes.WrapTx))
 
 	// Register services with Polaris.
-	p.RegisterServices(clientCtx, []node.Lifecycle{
+	p.RegisterLifecycles([]node.Lifecycle{
 		p.WrappedTxPool,
 	})
 
-	//
-	ek.StartEnginePlugin(clientCtx)
-	return nil
+	// Register the sync status provider with Polaris.
+	p.RegisterSyncStatusProvider(comet.NewSyncProvider(clientCtx))
+
+	// Start the services. TODO: move to place race condition is solved.
+	return p.StartServices()
 }
 
 // RegisterServices is a function that allows for the application to register lifecycles with
 // the evm networking stack. It takes a client context and a slice of node.Lifecycle
 // as arguments.
-func (p *Polaris) RegisterServices(_ client.Context, lcs []node.Lifecycle) {
+func (p *Polaris) RegisterLifecycles(lcs []node.Lifecycle) {
 	// Register the services with polaris.
 	for _, lc := range lcs {
-		p.RegisterService(lc)
+		p.ExecutionLayer.RegisterLifecycle(lc)
 	}
+}
 
-	// Start the services.
-	if err := p.StartServices(); err != nil {
-		panic(err)
-	}
+// StartServices starts the services of the Polaris struct.
+func (p *Polaris) StartServices() error {
+	go func() {
+		// TODO: these values are sensitive due to a race condition in the json-rpc ports opening.
+		// If the JSON-RPC opens before the first block is committed, hive tests will start failing.
+		// This needs to be fixed before mainnet as its ghetto af. If the block time is too long
+		// and this sleep is too short, it will cause hive tests to error out.
+		time.Sleep(5 * time.Second) //nolint:gomnd // as explained above.
```

Comment:
for now could just make this sleep time a function of the block time (like `2 * p.config.BlockTime` or so)

Reply:
gonna fix properly next PR